File: PlaylistObjC/PlaylistObjC/Playlist.m
//
//  Playlist.m
//  PlaylistObjC
//
//  Created by James Pacheco on 4/21/16.
//  Copyright © 2016 DevMountain. All rights reserved.
//

#import "Playlist.h"

@interface Playlist ()

@property (nonatomic, strong) NSMutableArray *internalSongs;

@end

@implementation Playlist

- (instancetype)initWithName:(NSString *)name songs:(NSArray *)songs
{
    self = [super init];
    if (self) {
        _name = name;
        _internalSongs = [songs mutableCopy];
    }
    return self;
}

#pragma mark - Properties

- (NSArray *)songs { return self.internalSongs; }

- (void)addSongsObject:(Song *)song;
{
	[self.internalSongs addObject:song];
}

- (void)removeSongsObject:(Song *)song;
{
	[self.internalSongs removeObject:song];
}

@end

File: Playlist/Playlist/Playlist.swift
//
//  Playlist.swift
//  Playlist
//
//  Created by James Pacheco on 5/4/16.
//  Copyright © 2016 DevMountain. All rights reserved.
//

import Foundation

class Playlist: Equatable {
    init(name: String, songs: [Song] = []) {
        self.name = name
        self.songs = songs
    }
	
	let name: String
	var songs: [Song]
}

func ==(lhs: Playlist, rhs: Playlist) -> Bool {
    return lhs.name == rhs.name && lhs.songs == rhs.songs
}